### PR TITLE
Improve the Raft upgrade procedure

### PR DESCRIPTION
**Pull request by kbr-**
Better logging, less code, a minor fix.

### CODE REVIEW

**Review by kostja — APPROVED** [review]

File: service/raft/raft_group0.cc [hunk 1]
```diff
@@ -1029,7 +1029,7 @@ struct sleep_with_exponential_backoff {
     static constexpr std::chrono::seconds _max_retry_period{16};
```
Comment (avikivity) [inline service/raft/raft_group0.cc:1029]: Unrelated tip: you can write `constexpr auto _max_retry_period = 16s;` for extra coolness.

File: service/raft/raft_group0.cc [hunk 2]
```diff
@@ -1027,7 +1027,9 @@ with_timeout(abort_source& as, db::timeout_clock::duration d, F&& fun) {
 struct sleep_with_exponential_backoff {
     std::chrono::seconds _retry_period{1};
     static constexpr std::chrono::seconds _max_retry_period{16};
-    future<> operator()(abort_source& as) {
+    future<> operator()(abort_source& as,
+                        std::experimental::source_location loc = std::experimental::source_location::current()) {
```
Comment (avikivity) [inline service/raft/raft_group0.cc:1031]: Please use std::source_location instead. See #12137

**Comment by kbr-** [comment]
v2: use `std::source_location`, without `experimental::`.

File: service/raft/raft_group0.cc (hunk 1 above)
Comment (kbr-) [reply service/raft/raft_group0.cc:1029]: Only with `using namespace std::chrono_literals` somewhere :(

**Comment by kbr-** [comment]
@scylladb/scylla-maint please merge

**Review by denesb — APPROVED** [review]

**Pull request closed without merging**